src/stemmablenote.ts

```diff
@@ -194,29 +194,27 @@ export abstract class StemmableNote extends Note {
 
   /** Gets the `y` value for the top modifiers at a specific `textLine`. */
   getYForTopText(textLine: number): number {
-    if (!this.stave) throw new Vex.RERR('NoStave', 'No stave attached to this note.');
     const extents = this.getStemExtents();
-    if (extents) {
+    if (this.hasStem()) {
       return Math.min(
-        this.stave.getYForTopText(textLine),
-        extents.topY - this.render_options.annotation_spacing * (textLine + 1)
+        this.stave?.getYForTopText(textLine) ?? 0,
```

Comment:
If you keep the `if (!this.stave)` above, you don't need the `stave?` or the `?? 0` -- these are silent failures.

Reply:
done